yt/frontends/enzo_p/data_structures.py

```diff
@@ -0,0 +1,353 @@
+"""
+Data structures for Enzo-P
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2017, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from yt.utilities.on_demand_imports import \
+    _h5py as h5py
+import numpy as np
+import os
+import stat
+
+from yt.funcs import \
+    ensure_tuple, \
+    get_pbar, \
+    setdefaultattr
+from yt.data_objects.grid_patch import \
+    AMRGridPatch
+from yt.data_objects.static_output import \
+    Dataset
+from yt.fields.field_info_container import \
+    NullFunc
+from yt.geometry.grid_geometry_handler import \
+    GridIndex
+from yt.utilities.logger import \
+    ytLogger as mylog
+
+from .fields import \
+    EnzoPFieldInfo
+
+from yt.frontends.enzo_p.misc import \
+    get_block_info, \
+    get_root_blocks, \
+    get_root_block_id
+
+class EnzoPGrid(AMRGridPatch):
+    """
+    Class representing a single EnzoP Grid instance.
+    """
+
+    def __init__(self, id, index, block_name, filename = None):
+        """
+        Returns an instance of EnzoPGrid with *id*, associated with
+        *filename* and *index*.
+        """
+        # All of the field parameters will be passed to us as needed.
+        AMRGridPatch.__init__(self, id, filename=filename, index=index)
+        self.block_name = block_name
+        self._children_ids = []
+        self._parent_id = -1
+        self.Level = -1
+
+    def __repr__(self):
+        return "EnzoPBlock_%s" % self.block_name
```

Comment:
this should match the name of the class, ideally for python objects exec(repr(obj)) should yield the object back (although that's probably not possible here)